src/core.h

```diff
@@ -17,40 +17,58 @@ class CVC;
 
 class Action {
   public:
+    Action(Character* actor, double score);
     //Get this action's actor (the character taking the action)
     Character* GetActor();
+    void SetActor(Character* actor);
+
     //Get the score this action's been given
     //it's assumed this is normalized against alternative actions
     //I'm not thrilled with this detail which makes it make sense ONLY in the
     //context of some other instances
     double GetScore();
+    void SetScore(double score);
 
     //Determine if this particular action is valid in the given gamestate by
     //the given character
-    virtual bool IsValid(CVC* gamestate) = 0;
+    virtual bool IsValid(const CVC& gamestate) = 0;
 
     //Have this action take effect by the given character
-    virtual void TakeEffect(CVC* gamestate) = 0;
+    virtual void TakeEffect(const CVC& gamestate) = 0;
```

Comment:
the game state WILL be changed in general here, although we're not going to reassign a new gamesate. That is, we're going to be updating the object in place (e.g.moving money from one character to another). is this the right use of const in that case?

Reply:
If gamestate will be modified, it shouldn't be (and can't be) const. This is more of an in/out parameter as oppose to a straight out parameter because you are just changing things on it.. but still shouldn't be const.